### PR TITLE
feat: add overridden_plan_id to plans and use it as default scope

### PR DESCRIPTION
**Pull request by lovrocolic**
## Context

Currently we are adding feature to allow overriding prices and charges while creating new subscription

## Description

This change adds required DB changes in order to support 'override prices and charges' feature.
Also, we created new scope that will not take into account any 'extended' plans in queries across the app

### CODE REVIEW

File: app/models/plan.rb [hunk 1]
```diff
@@ -24,6 +26,8 @@ class Plan < ApplicationRecord
   validates :code, presence: true, uniqueness: { scope: :organization_id }
   validates :amount_currency, inclusion: { in: currency_list }
 
+  scope :default, -> { where(overridden_plan_id: nil) }
```
Comment (rsempe) [inline app/models/plan.rb:29]: `default_scope` instead?

Comment (lovrocolic) [reply app/models/plan.rb:29]: I can switch to default scope but I personally don't like it since it mask behaviour :/

Comment (rsempe) [reply app/models/plan.rb:29]: I tend to agree 😅 

But I think we can use a different name than `default` then, something more explicit?

Comment (lovrocolic) [reply app/models/plan.rb:29]: `root`? 😃

Comment (vincent-pochet) [reply app/models/plan.rb:29]: `parent` makes more sense I think.

`default` could means many many things :smile:

**Review by vincent-pochet — COMMENTED** [review]
Good job! :rocket: 

Maybe it also worth to add at list one test case somewhere

File: app/models/plan.rb (hunk 1 above)
Comment (jdenquin) [reply app/models/plan.rb:29]: what about `base` ?

**Review by jdenquin — CHANGES_REQUESTED** [review]

File: app/models/plan.rb (hunk 1 above)
Comment (lovrocolic) [reply app/models/plan.rb]: switched to `base`

**Comment by lovrocolic** [comment]
> Good job! 🚀
> 
> Maybe it also worth to add at list one test case somewhere

Added one test in controller:)

**Review by vincent-pochet — APPROVED** [review]

File: app/models/plan.rb [hunk 2]
```diff
@@ -24,6 +26,8 @@ class Plan < ApplicationRecord
   validates :code, presence: true, uniqueness: { scope: :organization_id }
   validates :amount_currency, inclusion: { in: currency_list }
 
+  scope :base, -> { where(overridden_plan_id: nil) }
```
Comment (rsempe) [inline app/models/plan.rb:29]: `not_overriden`? 😅

File: app/models/plan.rb (hunk 1 above)
Comment (rsempe) [reply app/models/plan.rb]: `not_overriden`, explicit 😅

**Pull request closed without merging**